pkg/prometheus/promcfg.go

```diff
@@ -1929,3 +1864,157 @@ func (cg *ConfigGenerator) generateRemoteWriteConfig(
+	}
+
+	return slice
+}
+
+func (cg *ConfigGenerator) AppendRuleFiles(slice yaml.MapSlice, ruleFiles []string, ruleSelector *metav1.LabelSelector) yaml.MapSlice {
+	if ruleSelector != nil {
+		ruleFilePaths := []string{}
+		for _, name := range ruleFiles {
+			ruleFilePaths = append(ruleFilePaths, rulesDir+"/"+name+"/*.yaml")
+		}
+		slice = append(slice, yaml.MapItem{
+			Key:   "rule_files",
+			Value: ruleFilePaths,
+		})
+	}
+
+	return slice
+}
+
+func (cg *ConfigGenerator) AppendServiceMonitorConfigs(
+	slices []yaml.MapSlice,
+	serviceMonitors map[string]*v1.ServiceMonitor,
+	apiserverConfig *v1.APIServerConfig,
+	store *assets.Store,
+	shards int32) []yaml.MapSlice {
+	sMonIdentifiers := make([]string, len(serviceMonitors))
+	i := 0
+	for k := range serviceMonitors {
+		sMonIdentifiers[i] = k
+		i++
+	}
+
+	// Sorting ensures, that we always generate the config in the same order.
+	sort.Strings(sMonIdentifiers)
+
+	for _, identifier := range sMonIdentifiers {
+		for i, ep := range serviceMonitors[identifier].Spec.Endpoints {
+			slices = append(slices,
+				cg.WithKeyVals("service_monitor", identifier).generateServiceMonitorConfig(
+					serviceMonitors[identifier],
+					ep, i,
+					apiserverConfig,
+					store,
+					shards,
+				),
+			)
+		}
+	}
+
+	return slices
+}
+
+func (cg *ConfigGenerator) AppendPodMonitorConfigs(
+	slices []yaml.MapSlice,
+	podMonitors map[string]*v1.PodMonitor,
+	apiserverConfig *v1.APIServerConfig,
+	store *assets.Store,
+	shards int32) []yaml.MapSlice {
+	pMonIdentifiers := make([]string, len(podMonitors))
+	i := 0
+	for k := range podMonitors {
+		pMonIdentifiers[i] = k
+		i++
+	}
+
+	// Sorting ensures, that we always generate the config in the same order.
+	sort.Strings(pMonIdentifiers)
+
+	for _, identifier := range pMonIdentifiers {
+		for i, ep := range podMonitors[identifier].Spec.PodMetricsEndpoints {
+			slices = append(slices,
+				cg.WithKeyVals("pod_monitor", identifier).generatePodMonitorConfig(
+					podMonitors[identifier], ep, i,
+					apiserverConfig,
+					store,
+					shards,
+				),
+			)
+		}
+	}
+
+	return slices
+}
+
+func (cg *ConfigGenerator) AppendProbeConfigs(
+	slices []yaml.MapSlice,
+	probes map[string]*v1.Probe,
+	apiserverConfig *v1.APIServerConfig,
+	store *assets.Store,
+	shards int32) []yaml.MapSlice {
+	probeIdentifiers := make([]string, len(probes))
+	i := 0
+	for k := range probes {
+		probeIdentifiers[i] = k
+		i++
+	}
+
+	// Sorting ensures, that we always generate the config in the same order.
+	sort.Strings(probeIdentifiers)
+
+	for _, identifier := range probeIdentifiers {
+		slices = append(slices,
+			cg.WithKeyVals("probe", identifier).generateProbeConfig(
+				probes[identifier],
+				apiserverConfig,
+				store,
+				shards,
+			),
+		)
+	}
+
+	return slices
+}
+
+func (cg *ConfigGenerator) AppendAdditionalScrapeConfigs(slice yaml.MapSlice, additionalScrapeConfigs []byte, shards int32) ([]yaml.MapSlice, error) {
```

Comment:
`slice` isn't used anywhere in the function. Did you mean to append to the input slice and pass `scrapeConfigs` to the function?

Reply:
Oh, I think that was just copy-paste mistake 😅